test/conformance/runtime/readiness_probe_test.go

```diff
@@ -132,3 +142,98 @@ func TestProbeRuntime(t *testing.T) {
 		}
 	}
 }
+
+// This test validates the behaviour of readiness probes *after* initial startup.
+// The current behaviour is not ideal: when a pod goes unready after startup
+// and there are no other pods in the revision we hang, potentially forever,
+// which may not be what a user wants.
+// See https://github.com/knative/serving/issues/10765.
+func TestProbeRuntimeAfterStartup(t *testing.T) {
+	t.Parallel()
+	clients := test.Setup(t)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   test.Readiness,
+	}
+
+	test.EnsureTearDown(t, clients, &names)
+	resources, err := v1test.CreateServiceReady(t, clients, &names, v1opts.WithReadinessProbe(
+		&corev1.Probe{
+			// This behaviour is only the case where periodSeconds=0, because probes
+			// are ignored after startup when periodSeconds>0
+			// See https://github.com/knative/serving/issues/10764.
+			PeriodSeconds: 0,
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Path: "/healthz",
+				},
+			},
+		}))
+	if err != nil {
+		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
+	}
+
+	t.Log("POST to /start-failing")
+	url := resources.Route.Status.URL.URL()
+	client, err := pkgtest.NewSpoofingClient(context.Background(),
+		clients.KubeClient,
+		t.Logf,
+		url.Hostname(),
+		test.ServingFlags.ResolvableDomain,
+		test.AddRootCAtoTransport(context.Background(), t.Logf, clients, test.ServingFlags.HTTPS))
+	if err != nil {
+		t.Fatalf("Failed to create spoofing client: %v", err)
+	}
+
+	url.Path = "/start-failing"
+	startFailing, err := http.NewRequest(http.MethodPost, url.String(), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := client.Do(startFailing); err != nil {
+		t.Fatalf("POST to /start-failing failed: %v", err)
+	}
+
+	url.Path = "/"
+	// When periodSeconds = 0 we expect that once readiness propagates
+	// there will be no ready pods, and therefore the request will time
+	// out. (see https://github.com/knative/serving/issues/10765).
+	if err := wait.PollImmediate(1*time.Second, readinessPropagationTime, func() (bool, error) {
+		startFailing, err := http.NewRequest(http.MethodGet, url.String(), nil)
+		if err != nil {
+			return false, err
+		}
+
+		// 5 Seconds is enough to be confident the request is timing out.
+		client.Client.Timeout = 5 * time.Second
```

Comment:
I think this (and the error shenanigans) could be simplified  by creating a `NewRequestWithContext` using a `context.WithTimeout` and then checking the context for being timed out below. Might potentially spare you the "hack" inside the retry checker (or might totally break if the spoof client doesn't handle context errors properly :joy:)

Reply:
will give that a go and see what happens

Reply:
.. yeah no dice, spoof [overrides the context](https://github.com/knative/serving/blob/main/vendor/knative.dev/pkg/test/spoof/spoof.go#L167) pretty much immediately :(

Reply:
It augments the context tho, so it should still work.

Minor though, we don't need to block in that regard.

Reply:
hm yeah, but.. I gave it a go and it blocked forever rather than timing out so _something_ doesn't work there 🤔 Ill put it on my list to see if I missed something obvious there